### PR TITLE
Propagate nesting and principal for child executions

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
_Please replace this text with a description of what this PR accomplishes._

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/898

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by kumare3** [comment]
@katrogan do we need this - https://github.com/flyteorg/flytepropeller/blob/master/pkg/controller/nodes/subworkflow/launchplan.go#L60

**Comment by katrogan** [comment]
@kumare3 oh, no - I'll make a pr

File: pkg/manager/impl/execution_manager.go [hunk 1]
```diff
@@ -367,6 +367,43 @@ func (m *ExecutionManager) setCompiledTaskDefaults(ctx context.Context, task *co
...
 }
 
+// Fetches inherited execution metadata including the parent node execution db model id and the source execution model id
+// as well as sets request spec metadata with the inherited principal and adjusted nesting data.
+func (m *ExecutionManager) getInheritedExecMetadata(ctx context.Context, requestSpec *admin.ExecutionSpec,
+	workflowExecutionID *core.WorkflowExecutionIdentifier) (parentNodeExecutionID uint, sourceExecutionID uint, err error) {
+	if requestSpec.Metadata == nil || requestSpec.Metadata.ParentNodeExecution == nil {
+		return parentNodeExecutionID, sourceExecutionID, nil
+	}
+	parentNodeExecutionModel, err := util.GetNodeExecutionModel(ctx, m.db, requestSpec.Metadata.ParentNodeExecution)
```
Comment (EngHabu) [inline pkg/manager/impl/execution_manager.go:377]: Does this query the DB? if so, is that needed? don't we already have the node execution id in `requestSpec.Metadata.ParentNodeExecution`

Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:377]: yes, because we need the model id to map the relationship

Comment (EngHabu) [reply pkg/manager/impl/execution_manager.go:377]: ah because we need the DB id (int?) to store?

**Review by EngHabu — APPROVED** [review]

File: pkg/manager/impl/execution_manager.go (hunk 1 above)
Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:377]: yeah! s/model/DB/g

**Pull request merged**